src/notebooks/controllers/controllerLoader.ts

```diff
@@ -120,10 +120,6 @@ export class ControllerLoader implements IControllerLoader, IExtensionSyncActiva
                         this.registration
                     );
                 }
-
-                // Wait for all interpreters to be loaded,
-                // This reduces flickering (changing controllers from one to another).
-                await this.interpreters.waitForAllInterpretersToLoad();
```

Comment:
This was the problem, we were waiting for all python envs to get discovered before creating the controllers